### PR TITLE
Revert "Reland: "Fix how Gradle resolves Android plugin" (#137115)"

### PR DESCRIPTION
**Pull request by camsim99**
This reverts commit f5ac225c8da7a73ca16801054b8d4901d556b0fd, i.e. https://github.com/flutter/flutter/pull/137115.

This is a continuation of https://github.com/flutter/flutter/pull/142266 that was redone based on feedback to make this easier to revert in the future. The exact steps I took to create this revert:

1. Revert commit noted above
2. Fix merge conflicts, that notably involved reverting some changes in https://github.com/flutter/flutter/pull/140744 ~and https://github.com/flutter/flutter/pull/141417~ (fixed my merge to avoid the second PR from being affected)
3. Delete `packages/flutter_tools/test/integration.shard/android_plugin_skip_unsupported_test.dart` as this was added in the commit noted above

cc @Gustl22 since I couldn't tag as a reviewer

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by camsim99** [comment]
@bartekpacia This seems to be conflicting with https://github.com/flutter/flutter/pull/141417. Was planning to undo the changes to the example apps here, but not sure if this is caused by something I did in the merge.

File: packages/flutter_tools/gradle/src/main/groovy/flutter.groovy [hunk 1]
```diff
@@ -256,30 +257,7 @@ class FlutterPlugin implements Plugin<Project> {
...
-            flutterVersionCode = "1"
-        }
-        extension.flutterVersionCode = flutterVersionCode
-
-        Object flutterVersionName = localProperties.getProperty("flutter.versionName")
-        if (flutterVersionName == null) {
-            flutterVersionName = "1.0"
-        }
-        extension.flutterVersionName = flutterVersionName
-
```
Comment (bartekpacia) [inline packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:282]: @camsim99 this code was added in #141417

Comment (camsim99) [reply packages/flutter_tools/gradle/src/main/groovy/flutter.groovy:282]: Ok adding this back

**Comment by camsim99** [comment]
Before merging, I will file an issue to add back functionality to support https://github.com/flutter/flutter/pull/140744!

**Review by reidbaker — APPROVED** [review]

**Comment by camsim99** [comment]
> Before merging, I will file an issue to add back functionality to support #140744!

https://github.com/flutter/flutter/issues/142487

**Pull request merged**